### PR TITLE
RecordIO options for reader

### PR DESCRIPTION
**Pull request by sebheitzmann**

### CODE REVIEW

File: recordio/file_reader.go
Comment (sebheitzmann) [inline recordio/file_reader.go:353]: Hi Thomas, 

I have implemented this option pattern to be able to set the buffer size. The default one is too big for my use case.

File: simpledb/compaction.go
Comment (sebheitzmann) [inline simpledb/compaction.go:32]: just a little syntax error reported by my static analyser

File: simpledb/sstable_manager.go
Comment (sebheitzmann) [inline simpledb/sstable_manager.go:129]: this one is a bug if there are no record in a "delete only" sstable.

File: recordio/recordio_test.go
Comment (thomasjungblut) [inline recordio/recordio_test.go:70]: do you mind to make this buffer factory an option too? this path now doesn't run the DirectIO check anymore

File: simpledb/recovery.go
Comment (thomasjungblut) [inline simpledb/recovery.go:245]: oof, well spotted. Do you think we should return an error here?

**Comment by thomasjungblut** [comment]
just two minor fixes, if you find some time. I'll merge right after :) Thank you!

File: recordio/recordio_test.go
Comment (sebheitzmann) [reply recordio/recordio_test.go:70]: The DirectIOFactory is in the fonction. So i Think that this does exactly the same thing as previously. Sorry, I don't understand our comment.

File: simpledb/recovery.go
Comment (sebheitzmann) [reply simpledb/recovery.go]: yes, modified.

File: recordio/recordio_test.go
Comment (sebheitzmann) [reply recordio/recordio_test.go:70]: maybe you want a WithFactory() option ? Is it what you think ?

**Comment by thomasjungblut** [comment]
yes

File: recordio/recordio_test.go
Comment (sebheitzmann) [reply recordio/recordio_test.go:70]: I added a IoFactory Interface. Is it ok for you ?

**Comment by sebheitzmann** [comment]
tests are ok. If you can merge it would be wonderfull.

File: recordio/recordio_test.go
Comment (thomasjungblut) [reply recordio/recordio_test.go:70]: yes, but you need to pass the DirectIO factory here, it's otherwise not testing it

Comment (sebheitzmann) [reply recordio/recordio_test.go]: Done

**Review by thomasjungblut — APPROVED** [review]
lgtm, just waiting for the test to finish

**Comment by thomasjungblut** [comment]
Thank you @sebheitzmann - much appreciated :)

**Comment by sebheitzmann** [comment]
is it possible to tag a version ?

**Comment by thomasjungblut** [comment]
absolutely, give me a few minutes

**Pull request merged**
